src/catalog/test_case.rs

```diff
@@ -119,14 +124,121 @@ impl TestCase {
     }
 }
 
+fn verify_are_documents_ordered(collection_key: Vec<String>, documents: &[Value]) -> Result<()> {
+    use crate::doc::Pointer;
+    let pointers = collection_key
+        .iter()
+        .map(|p| Pointer::from(p))
+        .collect::<Vec<_>>();
+
+    let is_sorted = is_sorted_by(documents, |a, b| Pointer::compare(&pointers, a, b));
+    if !is_sorted {
+        let mut sorted = documents.to_vec();
+        // This is a stable sort, so the suggested order will change what's given as minimally as
+        // possible.
+        sorted.sort_by(|a, b| Pointer::compare(&pointers, a, b));
+        Err(TestExpectationsOutOfOrder {
+            collection_key,
+            reordered_documents: sorted,
+        })?
+    }
+    Ok(())
+}
+
+fn is_sorted_by<T, F>(seq: &[T], cmp_fun: F) -> bool
+where
+    F: Fn(&T, &T) -> std::cmp::Ordering,
+{
+    for (i, item) in seq.iter().enumerate() {
+        if let Some(n) = seq.get(i + 1) {
+            if cmp_fun(item, n) == std::cmp::Ordering::Greater {
+                return false;
+            }
+        } else {
+            return true;
+        }
+    }
+    true
+}
+
+#[derive(Debug)]
+pub struct TestExpectationsOutOfOrder {
+    collection_key: Vec<String>,
+    reordered_documents: Vec<Value>,
+}
+
+impl Display for TestExpectationsOutOfOrder {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        writeln!(f, "The expected documents must be provided in lixicographic order according to the collection key: [{}]",
```

Comment:
lexicographic. Also, "Documents to *verify* must... ?"

"expected" isn't a formal concept here; it's "verify" (unless we want to change it wholesale). Also please change TestExpectationsOutOfOrder to TestVerifyOutOfOrder (or similar) ? thx